robyn/templating.py

```diff
@@ -1,22 +1,84 @@
+from typing import Callable, List
 from abc import ABC, abstractmethod
 
 from jinja2 import Environment, FileSystemLoader
 
-from robyn import status_codes
+from robyn import status_codes, Robyn
+from robyn.router import Route
+# from robyn import robyn
 
 from .robyn import Headers, Response
 
 
+def get_param_filled_url(url: str, kwdict: dict | None = None) -> str:
+    """fill the :params in the url
+
+    Args:
+        url (str): typically comes from the route
+        kwdict (dict): the **kwargs as a dict
+
+    Returns:
+        str: _description_modified url (if there are elements in kwdict, otherwise unchanged)
+    """
+    if kwdict is not None:
+        for k, v in zip(kwdict.keys(), kwdict.values()):
+            url = url.replace(f":{k}", f"{v}")
+
+    return url
+
+
 class TemplateInterface(ABC):
     def __init__(self): ...
 
     @abstractmethod
     def render_template(self, *args, **kwargs) -> Response: ...
 
+    @abstractmethod
+    def set_robyn(self, robyn: Robyn) -> None: ...
+
+    @abstractmethod
+    def get_function_url(self, function_name: str, route_type: str = "GET", **kwargs) -> str: ...
```

Comment:
I put it in the ABC because I assume that if you switched from Jinja2 to a different templating engine you would still want to be able to call this function so you can have less brittle templates (ones that change if you change a subrouter prefix or the route for a function.